hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -80,7 +84,7 @@ func (tv *TypeVisitor) VisitDefinition(td TypeDefinition, ctx interface{}) (*Typ
 		return nil, errors.Wrapf(err, "visit of type of %q failed", td.Name())
 	}
 
-	def := MakeTypeDefinition(name, visitedType).WithDescription(td.description)
+	def := td.WithName(name).WithType(visitedType)
```

Comment:
Check that this has the same semantics - I have a (vague!) recollection that we wrote it using `MakeTypeDefinition()` to ensure the new type didn't carry over debris.

Reply:
I had to change it to ensure we do carry over the debris 😆

Reply:
actually it is not needed any more, but it's a bit simpler to preserve the description this way and less likely to break if we add anything else to `TypeDefinition` in future